battenberg/core.py

```diff
@@ -46,6 +47,8 @@ def is_installed(self) -> bool:
     def _fetch_remote_template(self):
         # First try to pull it from the remote origin/TEMPLATE_BRANCH
         keypair = construct_keypair()
+        if TEMPLATE_BRANCH not in list(self.repo.branches.local):
+            raise LocalBranchNotFoundException(f'{TEMPLATE_BRANCH} branch not found')
```

Comment:
This probably isn't what we want to do in this case. What the `fetch` line below is trying to do is to pull in the `template` branch from the remote if it doesn't find it locally, it just so happens in your case when it tries to do that it fails for a different error; that it does not have the SSH keypair configured correctly to be able to communicate with the remote repo.

Reply:
That's why I posted a Dockerfile in the issue so we could reproduce.
SSH keypair it's what I thought at first, but I can clone (using pygit2 directly) repo or use the `battenberg install` command for instance just fine, so I'm not sure it is related.

I tried not having ssh key or not having the ssh key authorized by gitlab, I got a different error message.

But if you delete the local branch template like so: https://github.com/zillow/battenberg/issues/26#issuecomment-880979917 , I was able to reproduce

I'm open for suggestion or maybe that's something that will need to be handled at the pygit lib level?